XiEditor/AppDelegate.swift

```diff
@@ -250,6 +283,22 @@ class AppDelegate: NSObject, NSApplicationDelegate, XiClient {
         testWindow?.contentView = TextPlaneDemo(frame: frame)
     }
 
+    var scrollTesters = [Document: ScrollTester]()
+    @IBAction func toggleScrollBenchmark(_ sender: AnyObject) {
```

Comment:
If this method lived in `EditViewController` you wouldn't have to do the dance of finding the correct document. You also wouldn't need the map, you could just have a `ScrollTester?` member variable in the view controller. (I'm also wary that using `Document`s as keys here could cause them to be retained if they're closed while this is running, which isn't a huge concern but could _possibly_ cause some debugging headache down the road)

Reply:
This would also potentially fix an issue I was having where the menu item wouldn't seem to do anything, or would start scrolling a background window.

Reply:
if I make the IBAction in EditViewController would Cocoa document magic automatically know the right document to talk to?

Reply:
ah I hadn't seen this. Yes: the foremost controller handles the selector, and it has a reference to the document.